lib/ansible/plugins/lookup/secret_server.py

```diff
@@ -0,0 +1,226 @@
+else:
+    SS_DOMAIN_ANSIBLE = None
+
+if os.getenv('SS_ADDRESS_ANSIBLE') is not None:
+    SS_ADDRESS_ANSIBLE = os.getenv('SS_ADDRESS_ANSIBLE')
+else:
+    SS_ADDRESS_ANSIBLE = None
+
+
+class SecretServer(object):
+    def __init__(self, username=None, password=None, secret_key=None,
+                 domain=None, address=None, key='password'):
+        self.username = username
+        self.password = password
+        self.secret_key = secret_key
+        self.domain = domain
+        self.address = address
+        self.key = key
+
+    def connect_api(self, path, address, auth_key=None, method='GET',
+                    connection='secure', headers=None, body=None):
+
+        if connection.upper() == 'SECURE':
+            connection = 'https://'
+        else:
+            connection = 'http://'
+
+        api_url = "{}{}" . format(connection, address)
+        url = "{}{}" . format(api_url, path)
+
+        if not headers:
+            headers = {
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Bearer' + ' ' + auth_key
+                    }
+        if method.upper() == 'GET':
+            r = requests.request(method, url, headers=headers, verify=False)
+        else:
+            r = requests.request(method, url, data=body, headers=headers)
+        return r.json()
+
+    def get_token(self):
+        user = self.username
+        password = self.password
+        domain = self.domain
+
+        if domain:
+            body = 'username={0}&password={1}&domain={2}&grant_type=password'.format(
+                    user, password, domain)
+        else:
+            body = 'username={0}&password={1}&grant_type=password'.format(
+                    user, password)
+        method = 'POST'
+        path = '/oauth2/token'
+        headers = {
+                'Content-Type': 'application/x-www-form-urlencoded',
+        }
+        api = self.connect_api(path, self.address, method=method, body=body,
+                               headers=headers)
+        if api.get('access_token'):
+            global token
+            token = api['access_token']
+        else:
+            raise AnsibleError("Secret Server Error: {0}".format(api))
+
+    def find_secret_id(self):
+        secret = self.secret_key
+
+        path = '/api/v1/secrets?filter.searchText={0}'.format(secret)
+        api = self.connect_api(path, self.address, auth_key=token)
+        if api.get('records'):
+            return api['records'][0]['id']
+        else:
+            raise AnsibleError("Secret Server Error: {0}".format(api))
+
+    def get_the_secret_value(self):
+        secret_id = self.find_secret_id()
+        key = self.key
+
+        path = '/api/v1/secrets/{0}/fields/{1}'.format(secret_id, key)
+        api = self.connect_api(path, self.address, auth_key=token)
+        return api
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
```

Comment:
We have recently made a decision that lookup should explicitly enumerate kwargs, instead of accepting splatted `**kwargs`

Reply:
that won't work with config values